### PR TITLE
Added cluster-group-upgrade package

### PR DESCRIPTION
**Pull request by shaior**

### CODE REVIEW

File: pkg/cgu/cgu.go
Comment (trewest) [inline pkg/cgu/cgu.go:175]:
```suggestion
	if err == nil {
		builder.Object = builder.Definition
	}
```

File: pkg/cgu/list.go
Comment (trewest) [inline pkg/cgu/list.go:14]:
```suggestion
func ListInAllNamespaces(apiClient *clients.Settings, options metaV1.ListOptions) ([]*CguBuilder, error) {
```

Comment (trewest) [inline pkg/cgu/list.go:14]: Please use the options parameter that is being provided

File: pkg/cgu/list.go
Comment (trewest) [inline pkg/cgu/list.go:17]:
```suggestion
	cguList := &v1alpha1.ClusterGroupUpgradeList{}
```

**Review by trewest — CHANGES_REQUESTED** [review]
Few comments

**Review by trewest — CHANGES_REQUESTED** [review]
Just one change and we can merge

**Comment by trewest** [comment]
@shaior can you resolve the merge conflicts? Other than that, looks good. Waiting for one more approval

**Review by cdvultur — CHANGES_REQUESTED** [review]
Small filename change, for consistency
Otherwise it looks great !

**Review by cdvultur — APPROVED** [review]
/lgtm

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**
